torch/_inductor/__init__.py

```diff
@@ -76,10 +78,13 @@ def list_mode_options(mode: str = None) -> Dict[str, Any]:
         "max-autotune-no-cudagraphs": {
             "max_autotune": True,
         },
-        # enable both cuda-graphs and max-autotune
+        # enable max-autotune
+        # enable cudagraphs when dynamic is not set
+        # otherwise, if both cudagraphs and dynamic are enabled, Inductor
+        # recompiles for each new shape
         "max-autotune": {
             "max_autotune": True,
-            "triton.cudagraphs": True,
+            "triton.cudagraphs": (dynamic is not True),
```

Comment:
Currently, we autotune based on the size hints, then use that config for all sizes.  This happens even if max_autotune is disabled (since we autotune other things like pointwise/reduction).  

IMO this is a decent heuristic and intentional behavior, though not perfect.  I don't think just disabling autotuning would be an improvement.

Reply:
max-autotune still works when dynamic is True, right? Only cudagraphs doesn't work. 
i.e. It will still try to search for different Triton configs and do gemm + epilogue fusions even with dynamic shapes. The input shape needs to be selected carefully (e.g. only use the max shape), but max-autotune would still work.
This would be the main use case in prod as well.

Reply:
Yes correct, max-autotune supports dynamic (by assuming the initial input is representative and tuning with that).